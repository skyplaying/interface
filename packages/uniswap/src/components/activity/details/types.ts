import {
  ConfirmedSwapTransactionInfo,
  ExactInputSwapTransactionInfo,
  ExactOutputSwapTransactionInfo,
  LpIncentivesClaimTransactionInfo,
  OffRampSaleInfo,
  OnRampPurchaseInfo,
  ReceiveTokenTransactionInfo,
  SendTokenTransactionInfo,
  TransactionType,
  TransactionTypeInfo,
  UnknownTransactionInfo,
} from 'uniswap/src/features/transactions/types/transactionDetails'

export type SwapTypeTransactionInfo =
  | ExactInputSwapTransactionInfo
  | ExactOutputSwapTransactionInfo
  | ConfirmedSwapTransactionInfo

export function isOnRampPurchaseTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is OnRampPurchaseInfo {
  return typeInfo.type === TransactionType.OnRampPurchase
}

export function isOffRampSaleTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is OffRampSaleInfo {
  return typeInfo.type === TransactionType.OffRampSale
}

export function isReceiveTokenTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is ReceiveTokenTransactionInfo {
  return typeInfo.type === TransactionType.Receive
}

export function isSendTokenTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is SendTokenTransactionInfo {
  return typeInfo.type === TransactionType.Send
}

export function isSwapTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is SwapTypeTransactionInfo {
  return typeInfo.type === TransactionType.Swap
}

export function isUnknownTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is UnknownTransactionInfo {
  return typeInfo.type === TransactionType.Unknown
}

export function isLpIncentivesClaimTransactionInfo(
  typeInfo: TransactionTypeInfo,
): typeInfo is LpIncentivesClaimTransactionInfo {
  return typeInfo.type === TransactionType.LPIncentivesClaimRewards
}

export type TransactionParticipantRowProps = {
  onClose: () => void
  address: string
  isSend?: boolean
}
