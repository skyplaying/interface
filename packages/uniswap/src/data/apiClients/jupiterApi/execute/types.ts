import { z } from 'zod'

// Types derived from https://dev.jup.ag/docs/api/ultra-api/execute

const StatusEnum = z.enum(['Success', 'Failed'])

const SwapEventSchema = z.object({
  inputMint: z.string(),
  inputAmount: z.string(),
  outputMint: z.string(),
  outputAmount: z.string(),
})

export const jupiterExecuteResponseSchema = z
  .object({
    status: StatusEnum,
    signature: z.string().optional(),
    slot: z.string().optional(),
    error: z.string().optional(),
    code: z.number(),
    totalInputAmount: z.string().optional(),
    totalOutputAmount: z.string().optional(),
    inputAmountResult: z.string().optional(),
    outputAmountResult: z.string().optional(),
    swapEvents: z.array(SwapEventSchema).optional(),
  })
  .passthrough()

export type JupiterExecuteResponse = z.infer<typeof jupiterExecuteResponseSchema>
