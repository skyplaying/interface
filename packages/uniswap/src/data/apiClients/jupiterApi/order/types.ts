import { Prettify } from 'viem'
import { z } from 'zod'

// Input types based on https://dev.jup.ag/docs/api/ultra-api/order, although some are missing from their documentation.
// Non-documented fields are discoverable via inspecting jup.ag's network requests.

export type JupiterOrderUrlParams = {
  inputMint: string
  outputMint: string
  amount: string
  taker?: string
  referralAccount?: string
  referralFee?: string
  slippageBps?: string
  swapMode: 'ExactIn' | 'ExactOut'
}

// Output types derived from https://dev.jup.ag/docs/api/ultra-api/order
// Some unused fields are commented out for potential future use.

const swapInfoSchema = z
  .object({
    ammKey: z.string(),
    label: z.string(),
    inputMint: z.string(),
    outputMint: z.string(),
    inAmount: z.string(),
    outAmount: z.string(),
    feeAmount: z.string(),
    feeMint: z.string(),
  })
  .passthrough()

const routePlanSchema = z.array(
  z
    .object({
      swapInfo: swapInfoSchema,
      percent: z.number(),
    })
    .passthrough(),
)

const platformFeeSchema = z
  .object({
    amount: z.string(),
    feeBps: z.number(),
  })
  .passthrough()

// const dynamicSlippageReportSchema = z
//   .object({
//     amplificationRatio: z.string().nullish(),
//     otherAmount: z.number().nullish(),
//     simulatedIncurredSlippageBps: z.number().nullish(),
//     slippageBps: z.number(),
//     categoryName: z.string(),
//     heuristicMaxSlippageBps: z.number(),
//   })
//   .passthrough()

const swapTypeSchema = z
  .union([
    z.literal('aggregator'),
    z.literal('rfq'),
    z.literal('hashflow'),
    z.string(), // allow unknown values too
  ])
  .transform((val) => {
    if (val === 'aggregator' || val === 'rfq' || val === 'hashflow') {
      return val
    }
    return 'unknown'
  })

export type JupiterOrderSwapType = z.infer<typeof swapTypeSchema>

export const jupiterOrderResponseSchema = z
  .object({
    inputMint: z.string(),
    outputMint: z.string(),
    inAmount: z.string(),
    outAmount: z.string(),
    otherAmountThreshold: z.string(),
    swapMode: z.enum(['ExactIn', 'ExactOut']),
    slippageBps: z.number(),
    priceImpactPct: z.string(),
    routePlan: routePlanSchema,
    feeMint: z.string().nullish(),
    feeBps: z.number(),
    prioritizationFeeLamports: z.number(),
    swapType: swapTypeSchema,
    transaction: z.string().nullish(),
    gasless: z.boolean(),
    requestId: z.string(),
    // totalTime: z.number(),
    taker: z.string().nullish(),
    quoteId: z.string().nullish(),
    maker: z.string().nullish(),
    // expireAt: z.string().nullish(),
    platformFee: platformFeeSchema.nullish(),
    // dynamicSlippageReport: dynamicSlippageReportSchema.nullish(),
    // errorMessage: z.string().nullish(),
  })
  .passthrough()

export type JupiterOrderResponse = Prettify<z.infer<typeof jupiterOrderResponseSchema>>
