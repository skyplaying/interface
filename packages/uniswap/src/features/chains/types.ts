// eslint-disable-next-line @typescript-eslint/no-restricted-imports
import { CurrencyAmount, Token, ChainId as UniswapSDKChainId } from '@uniswap/sdk-core'
import type { ImageSourcePropType } from 'react-native'
import { Chain as BackendChainId } from 'uniswap/src/data/graphql/uniswap-data-api/__generated__/types-and-hooks'
import { Platform } from 'uniswap/src/features/platforms/types/Platform'
import { ElementName } from 'uniswap/src/features/telemetry/constants'
import { NonEmptyArray } from 'utilities/src/primitives/array'
import { Chain as WagmiChain } from 'wagmi/chains'
// eslint-disable-next-line @typescript-eslint/no-restricted-imports
import { type UNIVERSE_CHAIN_INFO } from 'uniswap/src/features/chains/chainInfo'

export enum UniverseChainId {
  Mainnet = UniswapSDKChainId.MAINNET,
  ArbitrumOne = UniswapSDKChainId.ARBITRUM_ONE,
  Avalanche = UniswapSDKChainId.AVALANCHE,
  Base = UniswapSDKChainId.BASE,
  Blast = UniswapSDKChainId.BLAST,
  Bnb = UniswapSDKChainId.BNB,
  Celo = UniswapSDKChainId.CELO,
  MonadTestnet = UniswapSDKChainId.MONAD_TESTNET,
  Optimism = UniswapSDKChainId.OPTIMISM,
  Polygon = UniswapSDKChainId.POLYGON,
  Sepolia = UniswapSDKChainId.SEPOLIA,
  Soneium = UniswapSDKChainId.SONEIUM,
  Unichain = UniswapSDKChainId.UNICHAIN,
  UnichainSepolia = UniswapSDKChainId.UNICHAIN_SEPOLIA,
  WorldChain = UniswapSDKChainId.WORLDCHAIN,
  Zksync = UniswapSDKChainId.ZKSYNC,
  Zora = UniswapSDKChainId.ZORA,
  Solana = 501000101,
}

export type UniverseChainIdByPlatform<T extends Platform> = ((typeof UNIVERSE_CHAIN_INFO)[UniverseChainId] & {
  platform: T
})['id']
export type EVMUniverseChainId = UniverseChainIdByPlatform<Platform.EVM>
export type SVMUniverseChainId = UniverseChainIdByPlatform<Platform.SVM>

export interface EnabledChainsInfo {
  chains: UniverseChainId[]
  gqlChains: GqlChainId[]
  defaultChainId: UniverseChainId
  isTestnetModeEnabled: boolean
}

export enum RPCType {
  Public = 'public',
  Private = 'private',
  PublicAlt = 'public_alternative',
  Interface = 'interface',
  Fallback = 'fallback',
  Default = 'default',
}

export enum NetworkLayer {
  L1 = 0,
  L2 = 1,
}

export interface RetryOptions {
  n: number
  minWait: number
  medWait: number
  maxWait: number
}

export type GqlChainId = Exclude<BackendChainId, BackendChainId.UnknownChain | BackendChainId.EthereumGoerli>

export interface BackendChain {
  chain: GqlChainId
  /**
   * Set to false if the chain is not available on Explore.
   */
  backendSupported: boolean
  /**
   * Used for spot token prices
   */
  nativeTokenBackendAddress: string | undefined
}

type ChainRPCUrls = { http: string[] }
export interface UniverseChainInfo extends WagmiChain {
  readonly id: UniverseChainId
  readonly platform: Platform
  readonly assetRepoNetworkName: string | undefined // Name used to index the network on this repo: https://github.com/Uniswap/assets/
  readonly backendChain: BackendChain
  readonly blockPerMainnetEpochForChainId: number
  readonly blockWaitMsBeforeWarning: number | undefined
  readonly bridge?: string
  readonly docs: string
  readonly elementName: ElementName
  readonly explorer: {
    name: string
    url: `${string}/`
    apiURL?: string
  }
  readonly rpcUrls: {
    [RPCType.Default]: ChainRPCUrls
    [RPCType.Private]?: ChainRPCUrls
    [RPCType.Public]?: ChainRPCUrls
    [RPCType.PublicAlt]?: ChainRPCUrls
    [RPCType.Interface]: ChainRPCUrls
    [RPCType.Fallback]?: ChainRPCUrls
  }
  readonly interfaceName: string
  readonly label: string
  readonly logo: ImageSourcePropType
  readonly nativeCurrency: {
    name: string // 'Goerli ETH',
    symbol: string // 'gorETH',
    decimals: number // 18,
    address: string // '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
    explorerLink?: string // Special override for native ETH explorer link
    logo: ImageSourcePropType
  }
  readonly networkLayer: NetworkLayer
  readonly pendingTransactionsRetryOptions: RetryOptions | undefined
  /** Override the default spot price stablecoin amount, e.g. for chains with low liquidity. */
  readonly spotPriceStablecoinAmountOverride?: CurrencyAmount<Token>
  readonly tokens: {
    /** An array of stablecoins for this chain -- the first item in the array is treated as a 'default' stablecoin for this chain. */
    stablecoins: NonEmptyArray<Token>
    USDC?: Token
    DAI?: Token
    USDT?: Token
  }
  readonly statusPage?: string
  readonly subblockTimeMs?: number // in milliseconds, used for subblock balance checks
  readonly supportsV4: boolean
  readonly urlParam: string
  readonly wrappedNativeCurrency: {
    name: string // 'Wrapped Ether',
    symbol: string // 'WETH',
    decimals: number // 18,
    address: string // '0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6'
  }
  readonly tradingApiPollingIntervalMs: number
}
