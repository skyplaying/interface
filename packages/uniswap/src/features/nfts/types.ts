import { Chain, IAmount, NftsQuery } from 'uniswap/src/data/graphql/uniswap-data-api/__generated__/types-and-hooks'

export type GQLNftAsset = NonNullable<
  NonNullable<NonNullable<NonNullable<NftsQuery['portfolios']>[0]>['nftBalances']>[0]
>['ownedAsset']

export type NFTItem = {
  name?: string
  description?: string
  contractAddress?: string
  tokenId?: string
  imageUrl?: string
  thumbnailUrl?: string
  imageDimensions?: { width: number; height: number }
  collectionName?: string
  isVerifiedCollection?: boolean
  floorPrice?: number
  ownerAddress?: string
  listPrice?: IAmount
  isSpam?: boolean
  chain?: Chain
}
