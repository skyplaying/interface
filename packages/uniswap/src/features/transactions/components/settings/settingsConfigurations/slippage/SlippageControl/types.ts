import type { SlippageSettingsProps } from 'uniswap/src/features/transactions/components/settings/settingsConfigurations/slippage/useSlippageSettings'

export type SlippageControlProps = SlippageSettingsProps
