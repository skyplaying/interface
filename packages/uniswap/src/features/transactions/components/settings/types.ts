import type { AppTFunction } from 'ui/src/i18n/types'
import type { FeatureFlags } from 'uniswap/src/features/gating/flags'
import type { FrontendSupportedProtocol } from 'uniswap/src/features/transactions/swap/utils/protocols'

export enum TransactionSettingId {
  SLIPPAGE = 'slippage',
  DEADLINE = 'deadline',
}

export type TransactionSettingConfig = {
  renderTitle: (t: AppTFunction) => string
  hideTitle?: boolean
  Description?: React.FunctionComponent
  /** The UI that is displayed on the right side of a settings row, e.g. a Switch. If `Screen` is also defined, pressing `Control` will navigate to the screen. */
  Control: React.FunctionComponent
  /** The UI that will render if `Control` is pressed. */
  Screen?: React.FunctionComponent
  /** If defined, an info icon will appear next to `Title`, that will open `InfoModal` on press. */
  InfoModal?: React.FunctionComponent<{ isOpen: boolean; onClose: () => void }>
  /** If defined and the `featureFlag` is disabled, this setting will not be displayed. */
  featureFlag?: FeatureFlags
  settingId?: TransactionSettingId
  renderTooltip?: (t: AppTFunction) => string
  /** Returns warning configuration if the setting should show a warning */
  Warning?: React.FunctionComponent
}

export interface TransactionSettingsState {
  customSlippageTolerance?: number
  customDeadline?: number
  selectedProtocols: FrontendSupportedProtocol[]
  slippageWarningModalSeen: boolean
  isV4HookPoolsEnabled: boolean
}

export type TransactionSettings = TransactionSettingsState & { autoSlippageTolerance?: number }
