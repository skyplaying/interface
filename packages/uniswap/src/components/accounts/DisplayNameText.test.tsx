import { DisplayNameText } from 'uniswap/src/components/accounts/DisplayNameText'
import { DisplayName, DisplayNameType } from 'uniswap/src/features/accounts/types'
import { SAMPLE_SEED_ADDRESS_1 } from 'uniswap/src/test/fixtures'
import { render } from 'uniswap/src/test/test-utils'

const unitagDisplayName: DisplayName = { name: 'luni', type: DisplayNameType.Unitag }
const ensDisplayName: DisplayName = { name: 'vitalik.eth', type: DisplayNameType.ENS }
const localDisplayName: DisplayName = { name: 'Wallet 1', type: DisplayNameType.Local }
const addressDisplayName: DisplayName = {
  name: SAMPLE_SEED_ADDRESS_1,
  type: DisplayNameType.Address,
}

describe(DisplayNameText, () => {
  it('renders unitag without error', () => {
    const tree = render(<DisplayNameText displayName={unitagDisplayName} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('renders ens display name without error', () => {
    const tree = render(<DisplayNameText displayName={ensDisplayName} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('renders local display name without error', () => {
    const tree = render(<DisplayNameText displayName={localDisplayName} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('renders address display name without error', () => {
    const tree = render(<DisplayNameText displayName={addressDisplayName} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
