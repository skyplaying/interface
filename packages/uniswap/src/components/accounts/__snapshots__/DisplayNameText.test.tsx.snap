// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayNameText renders address display name without error 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    allowFontScaling={true}
    maxFontSizeMultiplier={1.4}
    numberOfLines={1}
    style={
      {
        "color": "#FFFFFF",
        "flexShrink": 1,
        "fontFamily": "Basel Grotesk",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22.1,
      }
    }
    suppressHighlighting={true}
  >
    0x82D56A352367453f74FC0dC7B071b311da373Fa6
  </Text>
</View>
`;

exports[`DisplayNameText renders ens display name without error 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    allowFontScaling={true}
    maxFontSizeMultiplier={1.4}
    numberOfLines={1}
    style={
      {
        "color": "#FFFFFF",
        "flexShrink": 1,
        "fontFamily": "Basel Grotesk",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22.1,
      }
    }
    suppressHighlighting={true}
  >
    vitalik.eth
  </Text>
</View>
`;

exports[`DisplayNameText renders local display name without error 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    allowFontScaling={true}
    maxFontSizeMultiplier={1.4}
    numberOfLines={1}
    style={
      {
        "color": "#FFFFFF",
        "flexShrink": 1,
        "fontFamily": "Basel Grotesk",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22.1,
      }
    }
    suppressHighlighting={true}
  >
    Wallet 1
  </Text>
</View>
`;

exports[`DisplayNameText renders unitag without error 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    allowFontScaling={true}
    maxFontSizeMultiplier={1.4}
    numberOfLines={1}
    style={
      {
        "color": "#FFFFFF",
        "flexShrink": 1,
        "fontFamily": "Basel Grotesk",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22.1,
      }
    }
    suppressHighlighting={true}
  >
    luni
  </Text>
  <View
    style={
      {
        "bottom": 2,
        "display": "inline",
        "flexDirection": "column",
        "transform": [
          {
            "translateY": 1,
          },
        ],
      }
    }
  />
</View>
`;
